src/main/scala/rocket/ICache.scala

```diff
@@ -54,7 +56,33 @@ class ICache(val icacheParams: ICacheParams, val hartId: Int)(implicit p: Parame
     name = s"Core ${hartId} ICache")))))
 
   val size = icacheParams.nSets * icacheParams.nWays * icacheParams.blockBytes
-  val device = new SimpleDevice("itim", Seq("sifive,itim0"))
+  val device = new SimpleDevice("itim", Seq("sifive,itim0")) {
+    override def getOMComponents(resourceBindingsMap: ResourceBindingsMap): Seq[OMComponent] = {
+      require(resourceBindingsMap.map.contains(this))
+      val resourceBindings = resourceBindingsMap.map.get(this)
+      resourceBindings.map {
+        rb => makeOMI(icacheParams, rb)
+      }.getOrElse(throw new NoSuchElementException)
+    }
+  }
+
+  def makeOMI(p: ICacheParams, resourceBindings: ResourceBindings): Seq[OMComponent] = {
```

Comment:
What is the rationale behind making this a separate function when it is only called once?

Reply:
Maybe this should be handled like the BTB, where it is a method on a helper object in the `diplomaticobjectmodel.model` package? I like that pattern

Reply:
Made the method on a helper object in the diplomaticobjectmodel.model package